src/pybind/mgr/mgr_module.py

```diff
@@ -409,19 +409,23 @@ def CLIWriteCommand(prefix: str, poll: bool = False) -> CLICommand:
     return CLICommand(prefix, "w", poll)
 
 
-def CLICheckNonemptyFileInput(func: HandlerFuncType) -> HandlerFuncType:
-    @functools.wraps(func)
-    def check(*args: Any, **kwargs: Any) -> Tuple[int, str, str]:
-        if 'inbuf' not in kwargs:
-            return -errno.EINVAL, '', ERROR_MSG_NO_INPUT_FILE
-        if isinstance(kwargs['inbuf'], str):
-            # Delete new line separator at EOF (it may have been added by a text editor).
-            kwargs['inbuf'] = kwargs['inbuf'].rstrip('\r\n').rstrip('\n')
-        if not kwargs['inbuf']:
-            return -errno.EINVAL, '', ERROR_MSG_EMPTY_INPUT_FILE
-        return func(*args, **kwargs)
-    check.__signature__ = inspect.signature(func)  # type: ignore[attr-defined]
-    return check
+def CLICheckNonemptyFileInput(content: str) -> HandlerFuncType:
+    def CheckFileInput(func: HandlerFuncType) -> HandlerFuncType:
+        @functools.wraps(func)
+        def check(*args: Any, **kwargs: Any) -> Tuple[int, str, str]:
+            if 'inbuf' not in kwargs:
+                return -errno.EINVAL, '', f'{ERROR_MSG_NO_INPUT_FILE} Please specify the file '\
+                                          f'containing {content} with "-i" option'
+            if isinstance(kwargs['inbuf'], str):
+                # Delete new line separator at EOF (it may have been added by a text editor).
+                kwargs['inbuf'] = kwargs['inbuf'].rstrip('\r\n').rstrip('\n')
+            if not kwargs['inbuf'].strip():
```

Comment:
Test failed with ```AttributeError: 'NoneType' object has no attribute 'strip'```:
```
______________________ SettingsTest.test_set_secret_empty ______________________

self = <dashboard.tests.test_settings.SettingsTest testMethod=test_set_secret_empty>

    def test_set_secret_empty(self):
>       r, out, err = handle_option_command(
            {'prefix': 'dashboard set-grafana-api-password'},
            None
        )

tests/test_settings.py:66: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
settings.py:243: in handle_option_command
    value, stdout, stderr = get_secret(inbuf=inbuf)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = (), kwargs = {'inbuf': None}

    @functools.wraps(func)
    def check(*args: Any, **kwargs: Any) -> Tuple[int, str, str]:
        if 'inbuf' not in kwargs:
            return -errno.EINVAL, '', f'{ERROR_MSG_NO_INPUT_FILE} Please specify the file '\
                                      f'containing {content} with "-i" option'
        if isinstance(kwargs['inbuf'], str):
            # Delete new line separator at EOF (it may have been added by a text editor).
            kwargs['inbuf'] = kwargs['inbuf'].rstrip('\r\n').rstrip('\n')
>       if not kwargs['inbuf'].strip():
E       AttributeError: 'NoneType' object has no attribute 'strip'
```

Reply:
Done